services/vision/mlvision/ml_model.go

```diff
@@ -0,0 +1,227 @@
+// Package mlvision uses an underlying model from the ML model service as a vision model,
+// and wraps the ML model with the vision service methods.
+package mlvision
+
+import (
+	"bufio"
+	"context"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/edaniels/golog"
+	"github.com/montanaflynn/stats"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/services/mlmodel"
+	"go.viam.com/rdk/services/vision"
+	"go.viam.com/rdk/utils"
+)
+
+var model = resource.NewDefaultModel("mlmodel")
+
+const (
+	// UInt8 is one of the possible input/output types for tensors.
+	UInt8 = "uint8"
+	// Float32 is one of the possible input/output types for tensors.
+	Float32 = "float32"
+)
+
+func init() {
+	resource.RegisterService(vision.Subtype, model, resource.Registration[vision.Service, *MLModelConfig]{
+		DeprecatedRobotConstructor: func(ctx context.Context, r any, c resource.Config, logger golog.Logger) (vision.Service, error) {
+			attrs, err := resource.NativeConfig[*MLModelConfig](c)
+			if err != nil {
+				return nil, err
+			}
+			actualR, err := utils.AssertType[robot.Robot](r)
+			if err != nil {
+				return nil, err
+			}
+			return registerMLModelVisionService(ctx, c.ResourceName().Name, attrs, actualR, logger)
+		},
+	})
+}
+
+// MLModelConfig specifies the parameters needed to turn an ML model into a vision Model.
+type MLModelConfig struct {
+	resource.TriviallyValidateConfig
+	ModelName string `json:"mlmodel_name"`
+}
+
+func registerMLModelVisionService(
+	ctx context.Context,
+	name string,
+	params *MLModelConfig,
+	r robot.Robot,
+	logger golog.Logger,
+) (vision.Service, error) {
+	_, span := trace.StartSpan(ctx, "service::vision::registerMLModelVisionService")
+	defer span.End()
+
+	mlm, err := mlmodel.FromRobot(r, params.ModelName)
+	if err != nil {
+		return nil, err
+	}
+	classifierFunc, err := attemptToBuildClassifier(mlm)
+	if err != nil {
+		logger.Infof("%v", errors.Wrapf(err, "was not able to turn ml model %q into a classifier", params.ModelName))
```

Comment:
based on my other example, please try to stick with using `<Level>w` type log invocations. It allows for structured logging which is easier to query wherever we export to. It also is consistent :)